pkg/reconciler/testing/factory.go

```diff
@@ -78,6 +80,11 @@ func MakeFactory(ctor Ctor) rt.Factory {
 		// set up Reconciler from fakes
 		r := ctor(t, ctx, &ls)
 
+		// promote the reconciler if it is leader aware
+		if la, ok := r.(reconciler.LeaderAware); ok {
+			la.Promote(reconciler.UniversalBucket(), func(reconciler.Bucket, types.NamespacedName) {})
+		}
```

Comment:
Consequence of 3.
Without it, the reconciliation is short-circuited because `reconciler.IsLeaderFor` always returns false (see code generated by injection-gen).